src/gpuarray_buffer_cuda.c

```diff
@@ -322,7 +322,8 @@ cuda_context *cuda_make_ctx(CUcontext ctx, gpucontext_props *p) {
   }
   res->errbuf->flags |= CUDA_MAPPED_PTR;
   /* Prime the cache */
-  cuda_alloc((gpucontext *)res, p->initial_cache_size, NULL, 0);
+  if (p->initial_cache_size)
```

Comment:
I forgot the detail, but why this don't cause a memory leak? We want to just fill the cache, not use it.